services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/client/ExtensionResolver.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.enhanced.dynamodb.internal.client;
+
+import java.util.Collections;
+import java.util.List;
+
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbEnhancedClientExtension;
+import software.amazon.awssdk.enhanced.dynamodb.extensions.VersionedRecordExtension;
+import software.amazon.awssdk.enhanced.dynamodb.internal.extensions.ChainExtension;
+
+/**
+ * Static module to assist with the initialization of an extension for a DynamoDB Enhanced Client based on supplied
+ * configuration.
+ */
+@SdkInternalApi
+public final class ExtensionResolver {
+    private static final DynamoDbEnhancedClientExtension DEFAULT_VERSIONED_RECORD_EXTENSION =
+        VersionedRecordExtension.builder().build();
+    private static final List<DynamoDbEnhancedClientExtension> DEFAULT_EXTENSIONS =
+        Collections.singletonList(DEFAULT_VERSIONED_RECORD_EXTENSION);
+
+    private ExtensionResolver() {
+    }
+
+    /**
+     * Static provider for the default extensions that are bundled with the DynamoDB Enhanced Client. Currently this is
+     * just the {@link software.amazon.awssdk.enhanced.dynamodb.extensions.VersionedRecordExtension}.
+     *
+     * These extensions will be used by default unless overridden in the enhanced client builder.
+     */
+    public static List<DynamoDbEnhancedClientExtension> defaultExtensions() {
+        return DEFAULT_EXTENSIONS;
```

Comment:
unmodifiable list?

Reply:
It's already immutable.

Reply:
As discussed I'll add a test to assert this and leave it at that.